### PR TITLE
EIP-7594: clarifying reconstruction and cross-seeding

### PR DESCRIPTION
**Pull request by cskiraly**
Attempt to clarify rules and recommendations for on-the-fly reconstruction and cross-seeding:
- make reconstruction a SHOULD if a node has enough data,
- allos delaying reconstruction,
- specify how to corss-seed reconstructed columns for subscribed and for unsubscribed topics.

Notes: For unsubscribed topics, we are not seeding to avoid potentially gererating too many duplicates. If, however, we want to speed up spread in this case, it might make sense to send it to a limited number (e.g. 1) of fanout peers, or to immediately generate IHAVE messages without waiting for the heartbeat.

### CODE REVIEW

**Comment by cskiraly** [comment]
@hwwhww , I have created this PR to improve the description of on-the-fly repair and related cross-seeding based on what we've discussed during the interop. I hope it maps well to the ideas that emerged. can you label this with eip-7594? Thanks!

File: specs/_features/eip7594/das-core.md [hunk 1]
```diff
@@ -235,11 +235,11 @@ Due to the deterministic custody functions, a node knows exactly what a peer sho
 
 ## Reconstruction and cross-seeding
 
-If the node obtains 50%+ of all the columns, they can reconstruct the full data matrix via `recover_matrix` helper.
+If a node fails to get a column on the column subnet, a node can also utilize the Req/Resp message to query the missing column from other peers.
```
Comment (fradamt) [inline specs/_features/eip7594/das-core.md:238]: This part feels like it belongs in some other section, it doesn't seem specific to reconstruction

Comment (cskiraly) [reply specs/_features/eip7594/das-core.md:238]: This paragraph was already in this subsection, just one sentence later. I agree we should move it somehere else. I will try to find a better place for it and move it.

**Review by nisdas — APPROVED** [review]

**Review by dapplion — APPROVED** [review]
New text sounds good me

**Review by ppopth — APPROVED** [review]
Looks good to me

**Pull request merged**